tangelo/linq/simulator_base.py

```diff
@@ -284,6 +398,54 @@ def _get_expectation_value_from_statevector(self, qubit_operator, state_prep_cir
 
         return expectation_value
 
+    def _get_variance_from_statevector(self, qubit_operator, state_prep_circuit, initial_statevector=None):
+        r"""Take as input a qubit operator H and a state preparation returning a
+        ket |\psi>. Return the variance <\psi | H | \psi>, computed
+        without drawing samples (statevector only). Users should not be calling
+        this function directly, please call "get_expectation_value" instead.
+
+        Args:
+            qubit_operator(openfermion-style QubitOperator class): a qubit
+                operator.
+            state_prep_circuit: an abstract circuit used for state preparation
+                (only pure states).
+
+        Returns:
+            complex: The variance of this operator with regards to the
+                state preparation.
+        """
+        n_qubits = state_prep_circuit.width
+
+        variance = 0.
+        prepared_frequencies, prepared_state = self.simulate(state_prep_circuit, return_statevector=True, initial_statevector=initial_statevector)
+
+        # Otherwise, use generic statevector expectation value
+        for term, coef in qubit_operator.terms.items():
+
+            if len(term) > n_qubits:  # Cannot have a qubit index beyond circuit size
+                raise ValueError(f"Size of operator {qubit_operator} beyond circuit width ({n_qubits} qubits)")
+            elif not term:  # Empty term: no simulation needed
+                variance += 0.
+                continue
+
+            if not self.n_shots:
+                # Without shots, variance using statevector is identically zero
+                variance += 0.
```

Comment:
If there are no shots we do nothing. So why don't we just test for the opposite case (`if self.n_shots`) and only do what's needed there? Then we can completely delete L431-433 I believe. 

It seems however further conversation may be needed about what this branch should actually do.